sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -178,6 +185,9 @@ class NodeHttpClient implements HttpClient {
 
       return response;
     } finally {
+      if (timeoutId !== undefined) {
+        clearTimeout(timeoutId);
```

Comment:
If the intent is the timeout only applies until we start receiving a response rather than when the response is fully received, can we move the clearTimeout up to right after the `makeRequest` promise returns the response object? 

Otherwise, we will not be consistent behaviorally between when the user returns a stream (and we clear the timeout before any response is streamed) and when we return bodyAsText (and we clear the timeout only after fully receiving the body and turning it into a string)

Reply:
Great point! Yes I think we should be consistent and not waiting for the whole body to be consumed

Reply:
Updated.